mlir/lib/Dialect/Rock/Generator/Conv2dGenerator.cpp

```diff
@@ -383,41 +350,35 @@ bool Conv2dGenerator::needExtraPadBwdWeight(OpBuilder &builder) const {
   GemmSize gemmSize = GemmSize::fromConvolution(dir, convDims);
 
   bool needExtraPad = false;
+  PopulateParamsInfo info{/*gemmSize=*/gemmSize,
+                          /*gemmFeatures=*/config.features,
+                          /*inputType=*/dataType,
+                          /*kernelType=*/KernelType::Conv2DBwdWeight,
+                          /*batchSize=*/convDims.n,
+                          /*numCu=*/static_cast<uint32_t>(config.num_cu)};
+
   if (!bitEnumContainsAll(config.features, GemmFeatures::mfma)) {
     PopulateParams populateParams;
     InitParamsNonXDL validParams;
-    // If there is a perfConfig present in the configuration
-    // we just need to see if padding will be used in that config
-    if (validParams.deserialize(config.perfConfig)) {
-      needExtraPad =
-          calculatePadding(validParams.gemmKPerBlock, validParams.gemmMPerBlock,
-                           validParams.gemmNPerBlock, gemmSize)
-              .has_value();
-    } else {
-      // This function will go through the list and pick a valid configuration
-      // the following code will return if the resulting config needs padding
-      needExtraPad =
-          calculatePaddingKernelSize(gemmSize, dir, dataType, populateParams)
-              .has_value();
-    }
+    uint32_t gridSize;
+    auto res = populateParams.obtainTuningParameters(info, 0, config.perfConfig,
+                                                     validParams, gridSize);
+    needExtraPad =
+        succeeded(res) &&
```

Comment:
So this translate to not having a perf_config str --> no need to pad. Is that always true ?

Reply:
About the perf_config, yes, it will go through the default list (ordered by padding). About exit the compilation when `obtainTuningParameters` fails, I don't have a strong opinion, but how do I do this? Do you mean adding an assert?

Reply:
AFAIK, this is how these are ultimately used... 

https://github.com/ROCmSoftwarePlatform/rocMLIR/blob/f432b86debf296cd32ce36384d2ee01bcc762dfa/mlir/tools/rocmlir-lib/rocmlir-lib.cpp#L132-L139

So we'd need to return nullptr ? 
@jerryyin thoughts ?

Reply:
You mean something like:
``` 
if (failed(conv2dGenerator.getWorkspaceSize(module))) { 
     return nullptr; 
} 
```
?

Reply:
Yea, something like that..

Reply:
(It is also used in kernel count as well...)

Reply:
Alright, I did it.